spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/actuate/GatewayControllerEndpoint.java

```diff
@@ -117,46 +114,25 @@ public void setApplicationEventPublisher(ApplicationEventPublisher publisher) {
 
 	// TODO: Flush out routes without a definition
 	@GetMapping("/routes")
-	public Mono<List<Map<String, Object>>> routes() {
-		Mono<Map<String, RouteDefinition>> routeDefs = this.routeDefinitionLocator
-				.getRouteDefinitions().collectMap(RouteDefinition::getId);
-		Mono<List<Route>> routes = this.routeLocator.getRoutes().collectList();
-		return Mono.zip(routeDefs, routes).map(tuple -> {
-			Map<String, RouteDefinition> defs = tuple.getT1();
-			List<Route> routeList = tuple.getT2();
-			List<Map<String, Object>> allRoutes = new ArrayList<>();
-
-			routeList.forEach(route -> {
-				HashMap<String, Object> r = new HashMap<>();
-				r.put("route_id", route.getId());
-				r.put("order", route.getOrder());
-
-				if (defs.containsKey(route.getId())) {
-					r.put("route_definition", defs.get(route.getId()));
-				}
-				else {
-					HashMap<String, Object> obj = new HashMap<>();
-
-					obj.put("predicate", route.getPredicate().toString());
-
-					if (!route.getFilters().isEmpty()) {
-						ArrayList<String> filters = new ArrayList<>();
-						for (GatewayFilter filter : route.getFilters()) {
-							filters.add(filter.toString());
-						}
-
-						obj.put("filters", filters);
-					}
-
-					if (!obj.isEmpty()) {
-						r.put("route_object", obj);
-					}
-				}
-				allRoutes.add(r);
-			});
-
-			return allRoutes;
-		});
+	public Flux<Map<String, Object>> routes() {
+		return this.routeLocator.getRoutes().map(this::serialize);
+	}
+
+	Map<String, Object> serialize(Route route) {
```

Comment:
I think we're missing the downstream URI here.

Reply:
Good catch.

Reply:
Fixed in 991a5d2db5bc83680fb82e3a30c12f59c323fb8b